core/src/io/anuke/mindustry/world/consumers/ConsumePower.java

```diff
@@ -72,7 +72,7 @@ public float requestedPower(TileEntity entity){
         if(buffered){
             return (1f-entity.power.satisfaction)*capacity;
         }else{
-            return usage;
+            return usage * entity.power.productivity;
```

Comment:
This is actually redundant; many blocks already override a method that specifies whether they should consume things: https://github.com/Quezler/Mindustry/blob/30a6506f736e0fee4fefda38e2776f607728384d/core/src/io/anuke/mindustry/world/blocks/units/UnitFactory.java#L197. I just never implemented it for power.

This can be changed to 
`usage * Mathf.num(entity.block.shouldConsume(entity.tile))`.

(Mathf.num should just convert a Boolean to 1 or 0, so if shouldConsume is false usage would be 0).

Reply:
done.